### ISSUE TITLE
Migrate to Gatsby v2

### ISSUE THREAD

**Issue by datakurre** [collective/gatsby-source-plone#83]
Gatsby v2 is so close that we should move to it as soon as possible and only target the plugin to v2. Maybe after the first evaluation?

https://github.com/gatsbyjs/gatsby/projects/2

https://v2--gatsbyjs.netlify.com/docs/migrating-from-v1-to-v2/

https://www.npmjs.com/package/gatsby?activeTab=versions

We can drop gatsby-plugin-sass from the example and let the example live without those styles until the plugin is compatible.

Opinions?

**Comment by ajayns** [comment]
I'm in total agreement to this, Gatsby is an active and fast changing community so it'd be a great idea stay up to date and targetting the latest. I'll do a bit of research, reading on this and after that we could think about making changes!

**Comment by cekk** [comment]
Absolutely +1
I've just read a post about it and wanted to open the same issue

**Comment by ajayns** [comment]
Going through the docs I think some actual experience migrating would help me get a better idea of doing, this, so I'm thinking of looking through and maybe helping out on this issue in the Gatsby project: https://github.com/gatsbyjs/gatsby/issues/5598

**Comment by ajayns** [comment]
So what are we giving priority to first? I'm still working on the recursive algorithm, but then do we migrate to v2 before we commence actual changes to add the recursive algorithm, or migrate after the plugin is working fully?

**Comment by datakurre** [comment]
After we have recursive traversal, V2 and images (files), we IMO should focus on making gh-pages to showcase the features and polish docs for a real release.

Things like mock content for schemas and support for collections are nice to haves that I would push only after the second coding period.

**Comment by datakurre** [comment]
https://www.gatsbyjs.org/blog/2018-06-16-announcing-gatsby-v2-beta-launch/

Nice to see @ajayns in v2 contributors :)

**Comment by ajayns** [comment]
Maybe I could add a bit of documentation about this and we could close this issue? I'm unsure though what could be included in it. Maybe I'll mention to migrating v1 to v2 guide?

**Comment by cekk** [comment]
I don't think that we need some kind of documentation about it.
We didn't published anything with v1 so the project will officially start with v2.

**Comment by ajayns** [comment]
Okay cool! @cekk that makes sense

**Comment by ajayns** [comment]
What else would be pending here then? Any steps more for this? Else we can go on and close this issue. @datakurre @cekk

**Comment by cekk** [comment]
I think that this could be closed. There is another issue with some problems.
